### PR TITLE
Open Studio sync after hosted site flow

### PR DESCRIPTION
**Pull request by sejas**
- Related to https://github.com/Automattic/dotcom-forge/issues/9359

## Proposed Changes

* Add logic to pass Studio site id and override redirection URL for new-hosted-site-flow

## Why are these changes being made?

* We want to use new-hosted-site-flow and at the end of the flow redirect to Studio

## Testing Instructions

Ideally the site id is a real Studio site from your appData.

* Follow this link: http://calypso.localhost:3000/setup/new-hosted-site/domains?studioSiteId=026a8080-362a-4ab9-921d-2c63510ffe5a
* Follow the steps and buy a site
* Observe the site is created and is atomic
* Observe the final URL is /home/siteId with the studioSiteId GET param: i.e. http://calypso.localhost:3000/home/239279206?studioSiteId=026a8080-362a-4ab9-921d-2c63510ffe5a
* Observe there is a notice and a popup triggered automatically

* Remove the studioSiteId param from the URL
* Observe the notice and alert are not visible

Final URL:
Screenshot 2024-11-22 at 22 19 49

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/customer-home/main.jsx
Comment (ivan-ottinger) [inline client/my-sites/customer-home/main.jsx:137]: I think you probably meant to trigger the redirect only if the `studioSiteId` query argument is present, right?

Because without the check, the redirection happens every time `siteId` changes - even when the Calypso Home is loaded without any relation to Studio connection.

**Review by ivan-ottinger — COMMENTED** [review]
The proposed changes look good! Great work on those redirects!

I noticed one issue that should be an easy fix though. Please see in the comment below. 🙂 

We could also consider updating the PR description to include URL without the `&showDomainStep` param - since we agreed on omitting it in the meantime.

(I could also make the changes to the PR if you would like to focus fully on your rotation)

File: client/my-sites/customer-home/main.jsx
Comment (sejas) [reply client/my-sites/customer-home/main.jsx:137]: Good catch! 🫶  Fixed https://github.com/Automattic/wp-calypso/pull/96691/commits/08bf0d639d7995881493ee8a56a6f9815da219ff

Comment (ivan-ottinger) [reply client/my-sites/customer-home/main.jsx:137]: Nice! Thank you for the fix, Antonio!

**Review by ivan-ottinger — APPROVED** [review]
I have given this PR a second review and test round and everything is looking good now. 👌🏼🙂 

The redirect happens as expected and I did not observe any regressions (even when testing the flow without the `studioSiteId` URL parameter).

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17023717

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @sejas for including a screenshot in the description! This is really helpful for our translators.

**Pull request merged**
